pkg/cloud/impl_registry.go

```diff
@@ -380,3 +472,91 @@ type ReadWriterInterceptor interface {
 	Reader(context.Context, ExternalStorage, ioctx.ReadCloserCtx) ioctx.ReadCloserCtx
 	Writer(context.Context, ExternalStorage, io.WriteCloser) io.WriteCloser
 }
+
+// makeExternalStorage creates an ExternalStorage from the given config.
+func makeExternalStorage[T any](
```

Comment:
nit: narrow the generic type? `s/T any/ T EarlyBootExternalStorageContext | ExternalStorageContext/r`

Reply:
Doneburger.